src/web/lib/components/Modal/index.js

```diff
@@ -9,6 +9,17 @@ export const Modal = ({ toggleModal, displayModal, children }) => {
     e.stopPropagation();
     toggleModal({ display: false });
   };
+
+  const handleKeyPress = event => {
+    if (event.keyCode === 27) {
+      toggleModal({ display: false });
+      console.log(event.keyCode);
+      document.removeEventListener("keydown", handleKeyPress);
+    }
+  };
+
+  document.addEventListener("keydown", handleKeyPress);
```

Comment:
You're adding a listener, and only removing it when the Esc key is pressed.

If the Esc key is not pressed, then the event listener will continue to exist, and over time the document will accumulate unused event listeners. As a result, when you press Esc, then `toggleModal` will be called multiple times, which may result in the lag that you've described.

Use `componentDidMount` and `componentWillUnmount` to correctly clean up the listeners, as I described in my comments in https://github.com/mozilla/FirefoxColor/issues/266

Reply:
Thanks for the guidance :)